Web/Edubase.Web.UI/Areas/Groups/Models/Validators/GroupEditorViewModelValidator.cs

```diff
@@ -104,6 +104,19 @@ public GroupEditorViewModelValidator(IGroupReadService groupReadService, IEstabl
                         .WithMessage("Closed date is invalid. Please enter a valid date.");
                     });
 
+                When(x => x.CanUserEditClosedDate
+                    && x.GroupType == eLookupGroupType.SecureSingleAcademyTrust
+                    && x.OriginalStatusId != (int) eLookupGroupStatus.Closed
+                    && x.StatusId == (int) eLookupGroupStatus.Closed
+                    && x.SaveGroupDetail, () =>
+                    {
+                        RuleFor(x => x.ClosedDate)
+                        .Must(x => !x.IsEmpty())
+                        .WithMessage("Please enter a date for the closure of this secure single-academy trust")
+                        .Must(x => x.IsValid() || x.IsEmpty())
+                        .WithMessage("Closed date is invalid. Please enter a valid date.");
+                    });
```

Comment:
What value does the `x.CanUserEditClosedDate` bit bring here?

_I appreciate that this is likely a copy/paste/tweak of the rule above so the original intention might not be clear, and the outcomes of this comment may apply to pre-existing code also! 😄_

My presumption is that it is to only validate fields the user may edit (reasonable), but here the validation centres around changing the status of the group.

Consider, for example the following scenario: 

- A user is configured to have permission to edit the group status, but not the close date
- Said user updates the status to closed
- ... here the validation (presumably) does not trigger because the user is not permitted to edit the closed date, and the group is then modified into an invalid state (closed but with a potentially invalid close date)

Maybe a mitigation is to ensure that permission to edit the status ALWAYS requires permission to edit the close date, but I would want to see this documented/enforced somewhere (perhaps in the API?).

Alternatively, my preferred option, we could just remove the edit close date condition and always require a valid close date regardless. If this edge case then results in a user with (mis?)configured permissions receiving an error they cannot correct, they would reach out and we can correct the permissions.

Reply:
@CathLass think this one is for you.

Reply:
In the FE, the permission to edit the status and the permission to edit the closed date are set to the same value:
            var userCanEditClosedDateAndStatus = UserCanEditClosedDateAndStatus(viewModel, user);
            if (userCanEditClosedDateAndStatus)
            {
                viewModel.CanUserEditClosedDate = true;
                viewModel.CanUserEditStatus = true;
            }
          
I think Roger's correct that the check x.CanUserEditClosedDate isn't needed here, as we are validating that the data is in a good state. Removing the check would cause a problem if the data had got into an invalid state elsewhere and a user hits this edit page without the permission to edit the closed date and status. I've removed it anyway.